meta-lmp-base/recipes-bsp/u-boot/u-boot-ostree-scr-fit/boot-common.cmd.in

```diff
@@ -37,6 +33,12 @@ if fiovb init ${devnum} && test -n "${board_is_closed}"; then
 	setenv fiovb_rpmb 1;
 else
 	echo "${fio_msg} Using ubootenv"
+	# Make sure basic environment is valid and call saveenv if not yet set
+	if test -z "${bootcount}"; then setenv bootcount 0; fi
+	if test -z "${rollback}"; then setenv rollback 0; fi
+	if test -z "${upgrade_available}"; then setenv upgrade_available 0; fi
+	if test -z "${bootupgrade_available}"; then setenv bootupgrade_available 0; run saveenv_mmc; fi
```

Comment:
this only saves the environment of bootupgrade_available was not set, does not address the other 3 variables

Reply:
Yeah, my idea was to use one variable as trigger, as we shouldn't be saving 4 times here (and bootcount could already be set to 0).

I decided to use bootupgrade_available as that is the only one that is "ours".

Reply:
From what I saw only rollback and bootupgrade_available are not set by u-boot (when comparing with machines setting their own standard / hardcoded environment and boot logic).